apps/dotcms-ui/src/app/portlets/dot-edit-page/components/dot-palette/dot-palette-contentlets/dot-palette-contentlets.component.ts

```diff
@@ -0,0 +1,161 @@
+    queryEsParams
+} from '@services/dot-paginator-es-content/dot-paginator-es-content.service';
+import { PaginatorService } from '@services/paginator';
+import { LazyLoadEvent } from 'primeng/api';
+import { take } from 'rxjs/operators';
+import { DotPaletteInputFilterComponent } from '../dot-palette-input-filter/dot-palette-input-filter.component';
+
+@Component({
+    selector: 'dot-palette-contentlets',
+    templateUrl: './dot-palette-contentlets.component.html',
+    styleUrls: ['./dot-palette-contentlets.component.scss']
+})
+export class DotPaletteContentletsComponent implements OnInit, OnChanges {
+    @ViewChild('filterInput', { static: true })
+    filterInput: DotPaletteInputFilterComponent;
+
+    @Input() contentTypeVariable: string;
+    @Input() languageId: string;
+    @Output() hide = new EventEmitter();
+    items: DotCMSContentlet[] | DotCMSContentType[] = [];
+    isFormContentType: boolean;
+    hideNoResults = true;
+
+    private _ESParams: queryEsParams;
+    private _filter: string;
+
+    constructor(
+        public paginatorESService: DotPaginatorESContentService,
+        public paginationService: PaginatorService,
+        private dotContentletEditorService: DotContentletEditorService
+    ) {}
+
+    ngOnInit(): void {
+        this._ESParams = {
+            itemsPerPage: 15,
+            lang: this.languageId || '1',
+            query: ''
+        };
+    }
+
+    ngOnChanges(changes: SimpleChanges): void {
+        if (changes?.contentTypeVariable?.currentValue) {
+            this.isFormContentType = changes?.contentTypeVariable?.currentValue === 'forms';
+
+            if (this.isFormContentType) {
+                this.paginationService.url = `v1/contenttype`;
+                this.paginationService.paginationPerPage = 15;
+                this.paginationService.sortField = 'modDate';
+                this.paginationService.setExtraParams('type', 'Form');
+                this.paginationService.sortOrder = 1;
+            }
+
+            this.loadData();
+        }
+    }
+
+    /**
+     * Loads data through pagination service
+     *
+     * @param LazyLoadEvent [event]
+     * @memberof DotPaletteContentletsComponent
+     */
+    loadData(event?: LazyLoadEvent): void {
+        if (this.isFormContentType) {
+            this.paginationService.setExtraParams(
+                'filter',
+                this.filterInput.searchInput.nativeElement.value
+            );
+            this.paginationService
+                .getWithOffset((event && event.first) || 0)
+                .pipe(take(1))
+                .subscribe((data: DotCMSContentType[] | DotCMSContentlet[]) => {
+                    data.forEach((item) => (item.contentType = item.variable = 'FORM'));
+                    this.items = data;
+                    this.hideNoResults = !!data?.length;
+                });
+        } else {
+            this.paginatorESService
+                .get({
+                    ...this._ESParams,
```

Comment:
Since this doesn't change, can't we just hardcodes here?

Is better for readability because you can see here in the request all the params you are sending.